src/model/services/mapi/corev1/createConfigMap.ts

```diff
@@ -0,0 +1,26 @@
+import { IOAuth2Provider } from 'lib/OAuth2/OAuth2';
+import { IHttpClient } from 'model/clients/HttpClient';
+import * as k8sUrl from 'model/services/mapi/k8sUrl';
```

Comment:
I guess there are reasons to not use some external k8s libs?

Reply:
The library scene for client-side k8s is pretty much unexistent at this time. Pretty much everything there is out there is for running on the server side.

We had some automation setup for generating all the client-side types and methods from our `apiextensions` repo [in here](https://github.com/giantswarm/giantswarm-cp-client). However, it was very brittle (something would always crash when adding new types), and the generated code was a bit clunky. Due to how it was built, we couldn't strip unused code at build time, so we had a huge bundle.

We reverted to using the K8s API as a regular REST API for now, and created a few utilities, such as `k8sUrl` to make it a bit easier

Reply:
How about this https://github.com/kubernetes-client/javascript ?

Reply:
That's only for the server-side.
https://github.com/kubernetes-client/javascript/issues/165

I tried porting it to the browser at some point, but there were so many Node.js native libraries to create, that I gave up eventually

Reply:
Alright I see what you mean by "client-side" now :)